mailbox/api/src/main/java/org/apache/james/mailbox/events/MailboxListener.java

```diff
@@ -309,12 +310,12 @@ public MailboxId getMailboxId() {
      * Indicates that mailbox has been deleted.
      */
     class MailboxDeletion extends MailboxEvent {
-        private final MailboxACL mailboxACL;
+        private final Optional<MailboxACL> mailboxACL;
```

Comment:
In the Java events we should enforce the ACLs to always be here IMO.

The serialization/deserialization layer should take care of providing sane defaults if none...